### PR TITLE
Add .spelling sort script

### LINKED ISSUE #22: .spelling file format
Discussion on how this should be formatted:
- categorised vs alphabetical
- 

There isn't any documentation anywhere for this.
This is adding extra complexity.
@Tyler-Ward @Adimote

### PR DESCRIPTION
**Pull request by Adimote**
add a script and rule to enforce that the .spelling file is sorted

resolves #22

### CODE REVIEW

File: scripts/sort-spelling.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
```
Comment (RealOrangeOne) [inline scripts/sort-spelling.sh]: This script seems unnecessary, running this manually is totally fine

Comment (trickeydan) [reply scripts/sort-spelling.sh]: I agree, especially as the CircleCI message refers to scripts/sort.sh rather than this file

**Comment by Adimote** [comment]
@RealOrangeOne @trickeydan please re-review as I've fixed the issues

File: scripts/sort.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
```
Comment (RealOrangeOne) [inline scripts/sort.sh]: This should be `/usr/bin/env bash`, as should the others

Comment (PeterJCLaw) [reply scripts/sort.sh]: Why? This script doesn't use any bashisms and `/bin/sh` _will_ exist.

Comment (RealOrangeOne) [reply scripts/sort.sh]: That's true, but i've always been told to use `env` for all scripts, _just in case_ it's used on a non-standard environment. Perhaps `/usr/bin/env sh` then?

Comment (PeterJCLaw) [reply scripts/sort.sh]: I think the encouragement around using `env` is valid for higher level things (`python`, `ruby` and the like) but shouldn't need to be the case for the shell itself.

I'm not sure whether `/bin/sh` is POSIX or not, though I think I'd ask why you think we can trust `/usr/bin/env` more than we can trust `/bin/sh`?

**Comment by kierdavis** [comment]
Surely just `sort --check .spelling` would accomplish more or less the same thing as this, in a far more maintainable way?

File: scripts/sort.sh (hunk 2 above)
Comment (kierdavis) [reply scripts/sort.sh]: I agree somewhat - I'd like us to get into the habit of using `/usr/bin/env` for script interpreters; in particular I'd prefer `/usr/bin/env bash` over `/bin/bash`, as the latter doesn't exist on my system.

I'm indifferent as to whether or not we do so for `sh`, however. My intuition suggests `/bin/sh` is more likely to exist on a hypothetical non-standard system than `/usr/bin/env`, but I have no data on this.

Comment (PeterJCLaw) [reply scripts/sort.sh]: Interesting, it looks like POSIX (explicitly) doesn't standardise this: https://en.wikipedia.org/wiki/Shebang_(Unix)#Portability.

My preference would still be for `/bin/sh` as I've never seen a system without it (though I have heard of systems without `/usr/bin/env`); however as @kierdavis points out in https://github.com/sourcebots/volunteer-docs/pull/28#issuecomment-375261727, `sort --check` renders this all moot.

**Comment by Adimote** [comment]
@kierdavis didn't know about that flag, that's much better!

File: scripts/sort.sh (hunk 2 above)
Comment (Adimote) [reply scripts/sort.sh]: Fixed by virtue of chaning to `sort --check` in 4db1f957c8fe4b9b9f8973ac3f8743513d0ace47

Comment (RealOrangeOne) [reply scripts/sort.sh]: Unfortunately changing the topic of this thread, do we really need this script now? It's trivial to reorder a file if it's incorrect, and the CI will check it for us. Most editors have the ability to sort a file alphabetically, so I don't think maintaining our own script is necessary?

Comment (kierdavis) [reply scripts/sort.sh]: A counterpoint: it's an advantage to have control over how the file is canonicalised. In particular, many editors won't remove duplicates or blank lines when sorting, which appears to be what we want here.

Comment (RealOrangeOne) [reply scripts/sort.sh]: The method of canonicalisation isn't important, however the check is. As the checking is in the CI, that should be good enough?

**Comment by Adimote** [comment]
So, funny story: It turns out `sort` is non-deterministic depending on what system you run it.

It also turns out that whatever system circleci runs it on has a different `sort` to mine, so when I ran `sort` it was sorting uppercase letters with lowercase, circleci was then checking it and saying it was wrong, because *it* expects it to be uppercase first then lowercase.

I've now changed a variable which now forces uppercase first

**Pull request merged**